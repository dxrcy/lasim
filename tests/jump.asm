.ORIG x3000

    and r1, r1, #0
    REG

    lea r1, X
    JMP r1

    and r1, r1, #0
    REG

X
    and r1, r1, #0
    add r1, r1, x0e
    REG

    JSR Foo

    and r1, r1, #0
    REG

    lea r1, Foo
    JSRR r1

    and r1, r1, #0
    REG

    HALT

Foo
    and r1, r1, #0
    add r1, r1, x03
    REG
    RET

.END
