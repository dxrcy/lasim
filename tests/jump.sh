#!/bin/sh

source "$(dirname $0)/shared.sh"

asm_file="$tests/jump.asm"
obj_file="$out/jump.obj"
output_actual_file="$out/jump.actual"
output_expected_file="$tests/jump.expected"

extract_reg() {
    sed -n 's/ *. *r1 *\([^ ]*\).*/\1/p'
}

lasim -a "$asm_file" -o "$obj_file"
lasim -x "$obj_file" | extract_reg > "$output_actual_file"

diff "$output_expected_file" "$output_actual_file"
report_status $?

