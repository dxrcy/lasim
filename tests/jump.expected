0x0000
0x000e
0x0003
0x0000
0x0003
0x0000
